synapse/app/homeserver.py

```diff
@@ -290,16 +289,6 @@ def start_listening(self) -> None:
                     manhole_settings=self.config.server.manhole_settings,
                     manhole_globals={"hs": self},
                 )
-            elif listener.type == "replication":
-                services = listen_tcp(
-                    listener.bind_addresses,
-                    listener.port,
-                    ReplicationStreamProtocolFactory(self),
```

Comment:
noting that this is still used in tests, so `ReplicationStreamProtocolFactory` isn't unreachable yet, hence reasonable to not remove it absolutely immediately in this PR